src/source/raster_dem_tile_worker_source.ts

```
@@ -6,43 +6,54 @@ import type {
     TileParameters
 } from './worker_source';
-import {isImageBitmap} from '../util/util';
+import {isImageBitmap, readImageUsingVideoFrame} from '../util/util';
+import {offscreenCanvasMangled} from '../util/offscreen_canvas_mangled';
+
+let offscreenCanvas: OffscreenCanvas;
+let offscreenCanvasContext: OffscreenCanvasRenderingContext2D;
 
 export class RasterDEMTileWorkerSource {
     actor: Actor;
     loaded: {[_: string]: DEMData};
-    offscreenCanvas: OffscreenCanvas;
-    offscreenCanvasContext: OffscreenCanvasRenderingContext2D;
 
     constructor() {
         this.loaded = {};
     }
 
-    loadTile(params: WorkerDEMTileParameters, callback: WorkerDEMTileCallback) {
+    async loadTile(params: WorkerDEMTileParameters, callback: WorkerDEMTileCallback) {
         const {uid, encoding, rawImageData, redFactor, greenFactor, blueFactor, baseShift} = params;
         // Main thread will transfer ImageBitmap if offscreen decode with OffscreenCanvas is supported, else it will transfer an already decoded image.
-        const imagePixels = isImageBitmap(rawImageData) ? this.getImageData(rawImageData) : rawImageData as RGBAImage;
+        const imagePixels = isImageBitmap(rawImageData) ? await this.getImageData(rawImageData) : rawImageData as RGBAImage;
         const dem = new DEMData(uid, imagePixels, encoding, redFactor, greenFactor, blueFactor, baseShift);
         this.loaded = this.loaded || {};
         this.loaded[uid] = dem;
         callback(null, dem);
     }
 
-    getImageData(imgBitmap: ImageBitmap): RGBAImage {
+    async getImageData(imgBitmap: ImageBitmap) {
+        const width = imgBitmap.width;
+        const height = imgBitmap.height;
+        const newWidth = width + 2;
+        const newHeight = height + 2;
+        if (offscreenCanvasMangled()) {
+            const parsed = await readImageUsingVideoFrame(imgBitmap, -1, -1, newWidth, newHeight);
+            if (parsed) return new RGBAImage({width: newWidth, height: newHeight}, parsed);
```

What happens if this is not parsed? do we return to "regular" flow? Should we throw an exception here?